internal/xds/translator/listener.go

```diff
@@ -401,7 +404,57 @@ func buildXdsDownstreamTLSSocket(tlsConfigs []*ir.TLSListenerConfig) (*corev3.Tr
 	}, nil
 }
 
-func buildXdsDownstreamTLSSecret(tlsConfig *ir.TLSListenerConfig) *tlsv3.Secret {
+func buildTLSParams(tlsConfig *ir.TLSListenerConfig) *tlsv3.TlsParameters {
+	p := &tlsv3.TlsParameters{}
+	isEmpty := true
+	if tlsConfig.MinVersion != nil {
+		p.TlsMinimumProtocolVersion = buildTLSVersion(tlsConfig.MinVersion)
+		isEmpty = false
+	}
+	if tlsConfig.MaxVersion != nil {
+		p.TlsMaximumProtocolVersion = buildTLSVersion(tlsConfig.MaxVersion)
+		isEmpty = false
+	}
+	if len(tlsConfig.Ciphers) > 0 {
+		p.CipherSuites = tlsConfig.Ciphers
+		isEmpty = false
+	}
+	if len(tlsConfig.ECDHCurves) > 0 {
+		p.EcdhCurves = tlsConfig.ECDHCurves
+		isEmpty = false
+	}
+	if len(tlsConfig.SignatureAlgorithms) > 0 {
+		p.SignatureAlgorithms = tlsConfig.SignatureAlgorithms
+		isEmpty = false
+	}
+	if isEmpty {
+		return nil
+	}
+	return p
+}
+
+func buildTLSVersion(version *gwv1a1.TLSVersion) tlsv3.TlsParameters_TlsProtocol {
+	lookup := map[gwv1a1.TLSVersion]tlsv3.TlsParameters_TlsProtocol{
+		gwv1a1.TLSv10: tlsv3.TlsParameters_TLSv1_0,
```

Comment:
thinking out loud, about the API, should the user strings be `1.2`, `1.3` instead of `v1_2` and `v1_3` ?

Reply:
I don't think it matters that much. I'll change to whatever is requested.

Reply:
FTR I think "1.2" and "1.3" are great. 🙂

Reply:
@liorokman sorry for churn, can you go ahead and update the strings to `1.2` & `1.3`

Reply:
Done.

One small thing to notice: when setting the TLS protocol version to the string `1.0`, sometimes YAML/JSON parsers detect the value as a number and not as a string and remove the `.0` from the value. It's important to quote the value in order to keep the `.0` in it as part of a string.

This can happen when the YAML text is unmarshaled into untyped maps. I ran into this when updating the YAML files that comprise the tests. It also happens with the tool that generates the CRD, which is why the validation enum comment contains quotes.

Example:
```yaml
spec:
    tls:
       # This sometimes gets parsed wrong, since it becomes a number and 
       # loses the fractional part.
       minVersion: 1.0
       # This always gets parsed correctly as a string.
       minVersion: "1.0"
```